packages/experimental/src/smart-session/controllers/SmartSessionsController.ts

```diff
@@ -230,5 +234,47 @@ export const SmartSessionsController = {
     } catch (e) {
       SnackController.showError('Error revoking smart session')
     }
+  },
+  async createSubscription(
+    request: CreateSubscriptionRequest
+  ): Promise<CreateSubscriptionResponse> {
+    let permissions: Permission[] = []
+
+    const interval = getIntervalInSeconds(request.interval)
+    const start = Date.now()
+    switch (request.asset) {
+      case 'native':
+        permissions = [
+          {
+            type: 'native-token-recurring-allowance',
+            data: {
+              allowance: request.amount,
+              start,
+              period: interval
+            }
+          }
+        ]
+        break
+      default:
+        throw new Error('Invalid asset')
+    }
+
+    return await this.grantPermissions({
+      chainId: request.chainId,
+      expiry: request.expiry,
+      signer: {
+        type: 'keys',
+        data: {
+          keys: [
+            {
+              type: 'secp256k1',
+              publicKey: request.signerPublicKey
+            }
+          ]
+        }
+      },
+      permissions,
+      policies: []
```

Comment:
Might users want to override `policies`? If so should it be received on params and passed here?